frontend/src/pages/projects/screens/projects/ProjectView.stories.tsx

```diff
@@ -0,0 +1,163 @@
+
+  // test that values from api are be displayed correctly
+  expect(await canvas.findByText('project', { selector: 'a' })).toBeInTheDocument();
+  expect(await canvas.findByText('admin')).toBeInTheDocument();
+};
+
+export const EditProject = Template.bind({});
+EditProject.parameters = {
+  a11y: {
+    // need to select modal as root
+    element: '.pf-c-backdrop',
+  },
+};
+EditProject.play = async ({ canvasElement }) => {
+  // load page and wait until settled
+  const canvas = within(canvasElement);
+  await canvas.findByTestId('story-loaded', undefined, { timeout: 5000 });
+  await waitForElementToBeRemoved(() => canvas.queryByText('Loading'), { timeout: 5000 });
+
+  // user flow for editing a project
+  await userEvent.click(canvas.getByLabelText('Actions', { selector: 'button' }));
+  await userEvent.click(canvas.getByText('Edit project', { selector: 'button' }));
+
+  // get modal
+  const body = within(canvasElement.ownerDocument.body);
+  const nameInput = body.getByRole('textbox', { name: /Name Resource name/i });
+  const updateButton = body.getByText('Update', { selector: 'button' });
+
+  // test that you can not submit on empty
+  await userEvent.clear(nameInput);
+  expect(updateButton).toBeDisabled();
+
+  // test that you can update the name to a different name
+  await userEvent.type(nameInput, 'Updated Project', { delay: 50 });
+  expect(updateButton).not.toBeDisabled();
+};
+
+export const DeleteProject = Template.bind({});
+DeleteProject.parameters = {
+  a11y: {
+    element: '.pf-c-backdrop',
+  },
+};
+DeleteProject.play = async ({ canvasElement }) => {
+  // load page and wait until settled
+  const canvas = within(canvasElement);
+  await canvas.findByTestId('story-loaded', undefined, { timeout: 5000 });
+  await waitForElementToBeRemoved(() => canvas.queryByText('Loading'), { timeout: 5000 });
+
+  // user flow for deleting a project
+  await userEvent.click(canvas.getByLabelText('Actions', { selector: 'button' }));
+  await userEvent.click(canvas.getByText('Delete project', { selector: 'button' }));
+
+  // get modal
+  const body = within(canvasElement.ownerDocument.body);
+  const retypeNameInput = body.getByRole('textbox');
+  const deleteButton = body.getByText('Delete project', { selector: 'button' });
+
+  // test that empty input disables form
+  expect(deleteButton).toBeDisabled();
+
+  // test that you can not submit on wrong input
+  await userEvent.type(retypeNameInput, 'not project', { delay: 50 });
+  expect(deleteButton).toBeDisabled();
+  await userEvent.clear(retypeNameInput);
+
+  // test that you can delete on correct input
+  await userEvent.type(retypeNameInput, 'project', { delay: 50 });
+  expect(deleteButton).not.toBeDisabled();
+};
+
+export const CreateProject = Template.bind({});
+CreateProject.parameters = {
+  a11y: {
+    element: '.pf-c-backdrop',
+  },
+};
+CreateProject.play = async ({ canvasElement }) => {
```

Comment:
Interesting -- going to need a crash course on what's going on here. Got links to docs to explain what is happening?

Since this is a common pattern, did we want a utility to setup a test?

---

`Template.bind` feels ... old. This is just a hack to duplicating an instance so you can mutate it?

Reply:
https://storybook.js.org/docs/6.5/react/writing-stories/args/

This talks about how a story is constructed. 

TLDR
The template is reused for each story.